### PR TITLE
Ks 316 replace secret usage in transmission protocol with the workflow execution

### PR DESCRIPTION
**Pull request by ettec**

### CODE REVIEW

**Review by cedric-cordenier — DISMISSED** [review]

File: core/capabilities/transmission/transmission.go [hunk 1]
```diff
@@ -49,9 +49,9 @@ func ExtractTransmissionConfig(config *values.Map) (TransmissionConfig, error) {
 // GetPeerIDToTransmissionDelay returns a map of PeerID to the time.Duration that the node with that PeerID should wait
 // before transmitting. If a node is not in the map, it should not transmit.  The sharedSecret is shared by nodes in the
 // same DON and used to generate a deterministic schedule for the transmission delays.
-func GetPeerIDToTransmissionDelay(donPeerIDs []ragep2ptypes.PeerID, sharedSecret [16]byte, transmissionID string, tc TransmissionConfig) (map[p2ptypes.PeerID]time.Duration, error) {
+func GetPeerIDToTransmissionDelay(donPeerIDs []ragep2ptypes.PeerID, transmissionID string, tc TransmissionConfig) (map[p2ptypes.PeerID]time.Duration, error) {
```
Comment (cedric-cordenier) [inline core/capabilities/transmission/transmission.go]: @ettec One minor comment here: I worry that by requiring callers to set transmission id correctly (to the wflow ID + execution ID) that we'll be encouraging misuse.

IMO it's better to require workflowID + workflowExecutionID to be passed in explicitly as variables named that way and then have the function add it; that should avoid the possibility of misuse (+ is nicely self-documenting too)

Comment (ettec) [reply core/capabilities/transmission/transmission.go]: sorted that, passed in the request rather than duplicate the extraction logic.

**Review by cedric-cordenier — DISMISSED** [review]

**Review by krehermann — DISMISSED** [review]

**Review by cedric-cordenier — APPROVED** [review]

**Review by krehermann — APPROVED** [review]

File: core/capabilities/transmission/transmission.go [hunk 2]
```diff
@@ -47,19 +49,29 @@ func ExtractTransmissionConfig(config *values.Map) (TransmissionConfig, error) {
...
+	tc, err := extractTransmissionConfig(req.Config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract transmission config from request: %w", err)
+	}
+
+	if req.Metadata.WorkflowID == "" || req.Metadata.WorkflowExecutionID == "" {
+		return nil, errors.New("workflow ID and workflow execution ID must be set in request metadata")
+	}
+
+	transmissionID := req.Metadata.WorkflowID + req.Metadata.WorkflowExecutionID
```
Comment (DeividasK) [inline core/capabilities/transmission/transmission.go:63]: WorkflowExecutionID = WorkflowID + EventID, so adding WorkflowID again seems redundant?

Comment (ettec) [reply core/capabilities/transmission/transmission.go:63]: agree, thats not necessary, no material impact I think? I'll get on to it shortly

**Pull request merged**